File: rust/src/models/fixed_window_reset_period.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::calendar_unit::CalendarUnit;

/// Resets at regular intervals — anchored to your subscription's exact activation time.
#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct FixedWindowResetPeriod {
    pub interval: i32,

    pub unit: CalendarUnit,
}

impl FixedWindowResetPeriod {
    pub fn new(interval: i32, unit: CalendarUnit) -> Self {
        Self { interval, unit }
    }
}

File: python/meteroid/models/fixed_window_reset_period.py
# this file is @generated
import dataclasses

from ..serialization import BaseModel
from .calendar_unit import CalendarUnit


@dataclasses.dataclass
class FixedWindowResetPeriod(BaseModel):
    """Resets at regular intervals — anchored to your subscription's exact activation time."""

    interval: int

    unit: CalendarUnit
